Geometry/HcalCommonData/src/HcalDDDRecConstants.cc

```diff
@@ -350,19 +362,20 @@ void HcalDDDRecConstants::initialize(void) {
 					  << " at index " << i 
 					  << " of etaTable from SimConstant";
     } else {
+      etaTable.push_back(hpar->etaTable[ieta]);
       etaSimValu.push_back(std::pair<int,int>(ef,ieta));
     }
-    for (int k=0; k<hpar->etagroup[i]; ++k) ietaMap.push_back(i+1);
+    for (int k=0; k<(hpar->etagroup[i]); ++k) ietaMap.push_back(i+1);
     if (ieta <= hpar->etaMax[0]) ietaHB = i+1;
     if (ieta <= hpar->etaMin[1]) ietaHE = i+1;
   }
-  assert( hpar->etaMin[1] == ietaHE );
-  assert( hpar->etaMax[0] == ietaHB );
+  iEtaMin[1] = ietaHE;
```

Comment:
The etamin and etamax at hcalparameter is set for Sim level and not at Reco level. The etatable and hence etamin and etamax could to be different (particularly for phaseII)
________________________________
From: Ianna Osborne [notifications@github.com]
Sent: 17 July 2015 07:51
To: cms-sw/cmssw
Cc: Sunanda Banerjee
Subject: Re: [cmssw] bsunanda:Run2-hcx22 Adopt simulation code to the new version of Hcal geometry (#10247)


In Geometry/HcalCommonData/src/HcalDDDRecConstants.cc<https://github.com/cms-sw/cmssw/pull/10247#discussion_r34863687>:

>      if (ieta <= hpar->etaMax[0]) ietaHB = i+1;
>      if (ieta <= hpar->etaMin[1]) ietaHE = i+1;
>    }
> -  assert( hpar->etaMin[1] == ietaHE );
> -  assert( hpar->etaMax[0] == ietaHB );
> +  iEtaMin[1] = ietaHE;


@bsunanda<https://github.com/bsunanda> - iEtaMin[1] == hpar->etaMin[1] and iEtaMax[0] == hpar->etaMax[0]

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/10247/files#r34863687>.